judgels-frontends/raphael/src/routes/uriel/contests/ContestsPage/ContestsPage.tsx

```diff
@@ -5,37 +5,89 @@ import { parse } from 'query-string';
 
 import Pagination from 'components/Pagination/Pagination';
 import { Card } from 'components/Card/Card';
+import SearchBox from 'components/SearchBox/SearchBox';
 import { Contest, ContestCreateData, ContestsResponse } from 'modules/api/uriel/contest';
 
 import { LoadingContestCard } from '../ContestCard/LoadingContestCard';
 import { ContestCard } from '../ContestCard/ContestCard';
 import { ContestCreateDialog } from '../ContestCreateDialog/ContestCreateDialog';
 import { contestActions as injectedContestActions } from '../modules/contestActions';
 
+import './ContestsPage.css';
+
 export interface ContestsPageProps extends RouteComponentProps<{ name: string }> {
   onGetContests: (name?: string, page?: number) => Promise<ContestsResponse>;
   onCreateContest: (data: ContestCreateData) => Promise<Contest>;
 }
 
+interface ContestsFilter {
+  name?: string;
+}
+
 export interface ContestsPageState {
   response?: ContestsResponse;
+  filter?: ContestsFilter;
+  isSearchBoxLoading?: boolean;
+  isContestsListLoading?: boolean;
```

Comment:
So, I have a better idea. Instead of using an additional flag like this, we can declare response to be:

```
response?: Partial<ContestsResponse>;
```

This way we can set `data` to be null/undefined, which is equivalent to the list being loading.